### PR TITLE
Typecast refactoring

### PR DESCRIPTION
**Pull request by Tigrov**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  |

### CODE REVIEW

**Review by terabytesoftw — APPROVED** [review]

**Review by darkdef — CHANGES_REQUESTED** [review]
1. Two type of changes in code: swap methods order and logical refactoring. Need split
2. match(true) - I think this bad practice and not approved this changes. Early exit work fastest.

**Comment by Tigrov** [comment]
> 1. Two type of changes in code: swap methods order and logical refactoring. Need split
> 2. match(true) - I think this bad practice and not approved this changes. Early exit work fastest.

1. Ok, agree. Thanks.
2. Why do you think `this bad practice`?
> Early exit work fastest.

In the case `=> $value` equals `return $value`

For me `match (true)` looks better than a lot of `if`
https://github.com/yiisoft/yii-queue/blob/master/src/Debug/QueueCollector.php#L37
https://github.com/yiisoft/log/blob/master/src/Message/Formatter.php#L148
https://github.com/yiisoft/yii-widgets/blob/master/src/Dropdown.php#L639

**Comment by darkdef** [comment]
> 2\. Why do you think `this bad practice`?

Match with short and simple condition - good solution (as examples from other packages), but in this case we have complex conditions.

File: tests/Provider/SchemaProvider.php [hunk 1]
```diff
@@ -303,7 +303,7 @@ public static function columnsTypeBit(): array
                         'size' => 1,
                         'precision' => 1,
                         'scale' => null,
-                        'defaultValue' => null,
+                        'defaultValue' => false,
```
[the remark this answers is not in the document]
Comment (Tigrov) [reply tests/Provider/SchemaProvider.php:306]: I think this line reduse it
https://github.com/yiisoft/db-mysql/pull/295/files#diff-aba1adae571d18ad5f7c6db19cfd8f15eb41daae1806924278c9a119f7c8b6daR564

Tests added

File: src/Schema.php [hunk 2]
```diff
@@ -488,15 +486,8 @@ protected function loadColumnSchema(array $info): ColumnSchemaInterface
...
-        $extra = $info['extra'];
-
-        if (str_starts_with($extra, 'DEFAULT_GENERATED')) {
-            $extra = strtoupper(substr($extra, 18));
-        }
-        $column->extra(trim($extra));
-
         if (preg_match('/^(\w+)(?:\(([^)]+)\))?/', $dbType, $matches)) {
-            $type = strtolower($matches[1]);
+            $type = $matches[1];
```
Comment (darkdef) [inline src/Schema.php:490]: Please return strtolower.
User can use constant PDO::CASE_UPPER in connection

**Review by darkdef — APPROVED** [review]

**Comment by darkdef** [comment]
Great work. Thanks

**Pull request merged**